File: 1200/1527B1/main.java
import java.util.*;
 
public class main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0)
        {
            int n=sc.nextInt();
            String s=sc.next();
            
            int zero=0;
            for(int i=0;i<s.length();i++)
            {
                if(s.charAt(i) == '0')
                    zero++;
            }
            
            if(zero == 1 || zero % 2 == 0) 
                System.out.println("BOB");
            else 
                System.out.println("ALICE");
        }
    }
}

File: 1200/1527B1/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;
        string s;
        cin >> s;

        int zero = 0;
        for (char c : s) 
        {
            if (c == '0') 
                zero++;
        }

        if (zero == 1 || zero % 2 == 0)
            cout << "BOB\n";
        else
            cout << "ALICE\n";
    }

    return 0;
}
